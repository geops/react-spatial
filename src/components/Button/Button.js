import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  /**
   * Function triggered on button's click event.
   */
  onClick: PropTypes.func.isRequired,

  /**
   * Children content of the button.
   */
  children: PropTypes.node.isRequired,

  /**
   * CSS class of the button.
   */
  className: PropTypes.string,

  /**
   * HTML tabIndex attribute
   */
  tabIndex: PropTypes.number,

  /**
   * Title of the button.
   */
  title: PropTypes.string,

  /**
   * HTML style attribute
   */
  style: PropTypes.object,
};

const defaultProps = {
  className: 'tm-button',
  title: undefined,
  tabIndex: undefined,
  style: undefined,
};

/**
 * This component displays a simple button.
 */
class Button extends PureComponent {
  render() {
    const { onClick, children, className, title, tabIndex, style } = this.props;

    return (
      <div
        role="button"
        style={style}
        tabIndex={tabIndex}
        className={className}
        title={title}
        aria-label={title}
        onClick={e => onClick(e)}
        onKeyPress={e => e.which === 13 && onClick(e)}
      >
        {children}
      </div>
    );
  }
}

Button.propTypes = propTypes;
Button.defaultProps = defaultProps;

export default Button;
