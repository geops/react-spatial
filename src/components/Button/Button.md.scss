.tm-button-example {
  cursor: pointer;
  color: $btn-primary-color;
  border-radius: 15px;
  width: 100px;
  text-align: center;
  border: 3px solid $btn-primary-color;
  padding: 10px;
  transition: background-color 0.1s ease;

  &:hover {
    background-color: $gray-lighter;
  }
}
