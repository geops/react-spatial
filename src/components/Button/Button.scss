.tm-button {
  cursor: pointer;
  height: $btn-size-base;
  width: $btn-size-base;
  display: flex;
  justify-content: center;
  align-items: center;
  transition: background-color 0.5s ease;
}

.tm-round-blue {
  background: $btn-secondary-color;
  border-radius: 50%;
  color: white;
}

.tm-square-white {
  border: 1px solid #aaa;
  background: white;
  color: #aaa;

  &.tm-active {
    background-color: $btn-secondary-color-active;
  }
}
