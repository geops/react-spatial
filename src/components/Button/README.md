#

This demonstrates the use of Button.

```js
import React from  'react'
import Button from 'react-spatial/components/Button';

<Button className="tm-button-example" onClick={() => alert('Clicked')}>
  Simple button
</Button>;
```
