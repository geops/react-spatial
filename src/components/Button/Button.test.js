import React from 'react';
import renderer from 'react-test-renderer';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Button from './Button';

configure({ adapter: new Adapter() });

const funcs = {
  onClick: () => {},
};

test('Button should match snapshot.', () => {
  const component = renderer.create(
    <Button
      className="tm-zoom in"
      title="Zoom in"
      onClick={() => funcs.onClick()}
    >
      +
    </Button>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Button should update.', () => {
  const spy = jest.spyOn(funcs, 'onClick');
  const bt = shallow(
    <Button className="tm-class" title="Zoom" onClick={() => funcs.onClick()}>
      +
    </Button>,
  );

  bt.find('.tm-class')
    .first()
    .simulate('click');
  bt.find('.tm-class')
    .first()
    .simulate('keypress', { which: 13 });
  expect(spy).toHaveBeenCalledTimes(2);
});
