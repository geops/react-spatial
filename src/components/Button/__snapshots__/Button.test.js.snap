// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match snapshot. 1`] = `
<div
  aria-label="Zoom in"
  className="tm-zoom in"
  onClick={[Function]}
  onKeyPress={[Function]}
  role="button"
  title="Zoom in"
>
  +
</div>
`;
